lib/openstack_api/openstack_identity.py

```diff
@@ -142,18 +142,64 @@ def find_user_all_domains(
         with self._connection_cls(cloud_account) as conn:
             return conn.identity.find_user(user_identifier, ignore_missing=True)
 
+    def _find_project_tag_index(
+        self, project_tags: List[str], select_func: Callable[[str], bool]
+    ) -> int:
+        """
+        Returns the index of a project tag matching a given query
+        :param project_tags: The project tags
+        :param select_func: Function that determines whether a tag should be selected
+        :return: The index of found tag or -1
+        """
+        for i, project_tag in enumerate(project_tags):
+            if select_func(project_tag):
+                return i
+        return -1
```

Comment:
So this is C-Style coding so slightly goes against being Pythonic

In this case can we simply:
```python
def _find_project_tag_index( ... ) -> Optional[int]:
    ....
        return i
    return None
```

Callers should be explicitly checking for `result is None` rather than `not result` explicitly with positional indices anyway to prevent `0 == false` bugs